File: programs/conditionals-logic_exercises-3.cpp
#include <iostream>

using namespace std;

// Assume isHoopy and isFrabjuous are defined elsewhere
bol isHoopy(int x);
bol isFrabjuous(int x);

int main() {
  bool flag1 = isHoopy(202);
  bool flag2 = isFrabjuous(202);
  cout << flag1 << endl;
  cout << flag2 << endl;
  if (flag1 && flag2) {
    cout << "ping!" << endl;
  }
  if (flag1 || flag2) {
    cout << "pong!" << endl;
  }
  return 0;
}

File: programs/conditionals-logic_exercises-3.java
public static void main(String[] args) {
    boolean flag1 = isHoopy(202);
    boolean flag2 = isFrabjuous(202);
    System.out.println(flag1);
    System.out.println(flag2);
    if (flag1 &amp;&amp; flag2) {
        System.out.println("ping!");
    }
    if (flag1 || flag2) {
        System.out.println("pong!");
    }
}